src/server/scripts/Events/midsummer.cpp

```diff
@@ -49,6 +69,319 @@ class go_midsummer_bonfire : public GameObjectScript
+
+                return;
+            }
+        }
+
+        player->RemoveAurasDueToSpell(SPELL_BONFIRES_BLESSING);
+    }
+};
+
+struct npc_midsummer_bonfire : public ScriptedAI
+{
+    npc_midsummer_bonfire(Creature* creature) : ScriptedAI(creature)
+    {
+        _isStampedOut  = nullptr;
+        _teamId     = TEAM_NEUTRAL;
+        _type       = BONFIRE_TYPE_NONE;
+        _spellFocus = nullptr;
+
+        if (!IsHolidayActive(HOLIDAY_FIRE_FESTIVAL))
+            return;
+
+        scheduler.Schedule(420ms, [this](TaskContext context)
+            {
+                if (!InitBonfire())
+                    context.Repeat();
+            });
+    }
+
+    void Ignite()
+    {
+        if (_isStampedOut)
+            *_isStampedOut = false;
+
+        if (!_spellFocus)
+        {
+            me->CastSpell(me, SPELL_MIDSUMMER_BONFIRE_BUNNIES_2, true);
+
+            if ((_spellFocus = me->FindNearestGameObject(GO_MIDSUMMER_BONFIRE_CAMPFIRE_SPELL_FOCUS, 10.0f)))
+                me->AddGameObject(_spellFocus);
+        }
+
+        switch (_type)
+        {
+            case BONFIRE_TYPE_ALLIANCE:
+            case BONFIRE_TYPE_HORDE:
+                me->CastSpell(me, SPELL_LIGHT_BONFIRE_ART_KIT, true);
+                UpdateBonfireBlessingBuffs();
+                break;
+            case BONFIRE_TYPE_AHUNE:
+                if (_bonfire)
+                    _bonfire->SetGoState(GO_STATE_ACTIVE);
+                break;
+            default:
+                break;
+        }
+    }
+
+    void StampOut()
+    {
+        switch (_type)
+        {
+            case BONFIRE_TYPE_ALLIANCE:
+            case BONFIRE_TYPE_HORDE:
+                if (_isStampedOut)
+                    *_isStampedOut = true;
+
+                if (_spellFocus)
+                {
+                    _spellFocus->DespawnOrUnsummon();
+                    _spellFocus = nullptr;
+                }
+
+                me->CastSpell(me, SPELL_STAMP_OUT_BONFIRE_ART_KIT, true);
+                UpdateBonfireBlessingBuffs();
+                break;
+            default:
+                break;
+        }
+    }
+
+    void UpdateBonfireBlessingBuffs()
+    {
+        if ((_type != BONFIRE_TYPE_ALLIANCE) && (_type != BONFIRE_TYPE_HORDE))
+            return;
+
+        me->GetMap()->DoForAllPlayers([&](Player* p)
+            {
+                if ((p->GetZoneId() == me->GetZoneId()) && (p->GetTeamId() == _teamId))
+                {
+                    if (_isStampedOut)
+                    {
+                        if (*_isStampedOut)
+                        {
+                            p->RemoveAurasDueToSpell(SPELL_BONFIRES_BLESSING);
+                        }
+                        else
+                        {
+                            if (!p->HasAura(SPELL_BONFIRES_BLESSING))
+                                p->CastSpell(p, SPELL_BONFIRES_BLESSING, true);
+                        }
```

Comment:
ok my man, at least choose one style of brackets, don't mix them 😅

Reply:
I usually do brackets if there's more than one statement, is this already "mixing styles"?

Reply:
also if the else clause has brackets the if should also have them? but whatever you prefer i can update it